src/Symfony/Component/Security/Acl/Permission/AbstractMaskBuilder.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Symfony\Component\Security\Acl\Permission;
+
+abstract class AbstractMaskBuilder implements MaskBuilderInterface
+{
+    /**
+     * @var int
+     */
+    protected $mask;
+
+    /**
+     * Constructor.
+     *
+     * @param int $mask optional; defaults to 0
+     */
+    public function __construct($mask = 0)
+    {
+        $this->set($mask);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function set($mask)
+    {
+        if (!is_int($mask)) {
+            throw new \InvalidArgumentException('$mask must be an integer.');
```

Comment:
I did not change this part, i just moved it into a set() method : https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Security/Acl/Permission/MaskBuilder.php#L82

But i can add an InvalidArgumentException under the Symfony\Component\Security\Acl\Exception if you think it's better